web/views/task_view.ex

```diff
@@ -1,14 +1,16 @@
 defmodule CoursePlanner.TaskView do
   @moduledoc false
   use CoursePlanner.Web, :view
+  alias CoursePlanner.{SharedView, Volunteers}
+  alias Ecto.Changeset
 
-  def task_user(conn, task) do
-    if task.user_id do
-      link task.user.name, to: volunteer_path(conn, :show, task.user)
-    else
-      "no one"
-    end
+  def task_user(conn, volunteers) when not is_nil(volunteers) do
+    volunteers
+    |> Enum.map(fn(volunteer) ->
+         link volunteer.name, to: volunteer_path(conn, :show, volunteer)
+       end)
   end
+  def task_user(_conn, _volunteers), do: "no one"
```

Comment:
is this ever reached? as far as I remember volunteers will be always a list, an empty list in case there is no volunteers